[Question] How to run mcp using NPX

i'm upload MCP serrer my private package store, 

and run below error occured

```
[2025-06-11T03:04:01.832Z] [INFO] Initializing MCP Server: @packge/my-packge@0.0.6
[2025-06-11T03:04:01.834Z] [INFO] Starting MCP server: (Framework: 0.2.13, SDK: 1.12.1)...
[2025-06-11T03:04:01.834Z] [INFO] Capabilities detected: {}
[2025-06-11T03:04:01.835Z] [ERROR] Server failed to start: Server does not support tools (required for tools/list)
Error: Server does not support tools (required for tools/list)
    at Server.assertRequestHandlerCapability (file:///Users/macbook-user/.npm/_npx/e4fe9340b75556e8/node_modules/@modelcontextprotocol/sdk/dist/esm/server/index.js:129:27)
    at Server.setRequestHandler (file:///Users/macbook-user/.npm/_npx/e4fe9340b75556e8/node_modules/@modelcontextprotocol/sdk/dist/esm/shared/protocol.js:324:14)
    at MCPServer.setupHandlers (file:///Users/macbook-user/.npm/_npx/e4fe9340b75556e8/node_modules/mcp-framework/dist/core/MCPServer.js:152:21)
    at MCPServer.start (file:///Users/macbook-user/.npm/_npx/e4fe9340b75556e8/node_modules/mcp-framework/dist/core/MCPServer.js:315:18)
```

how can i resolve it ?

here is my directory structure

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/bbbd0245-7684-4bfc-bb22-b2d42b161286" />

and here is my package.json

```

  "type": "module",
  "bin": {
    "my-mcp-server": "./dist/index.js"
  },
  "files": [
    "dist",
    "package.json"
  ],
  
  "scripts": {
    "build": "rm -rf ./dist && tsc && mcp-build && chmod +x dist/index.js",
    "inspect": "rm -rf ./dist && tsc && mcp-build && chmod +x dist/index.js && npx @modelcontextprotocol/inspector dist/index.js",
    "watch": "tsc --watch",
    "start": "node dist/index.js",
    "pub": "npm run build && npm link"
  },
  "dependencies": {
    "@modelcontextprotocol/sdk": "^1.8",
    "axios": "^1.9.0",
    "mcp-framework": "^0.2.11",
    "module-alias": "^2.2.3"
  },
  "devDependencies": {
    "@types/node": "^20.11.24",
    "typescript": "^5.3.3"
  },
  "engines": {
    "node": ">=22.14.0"
  }
```

--- a/src/core/MCPServer.ts
+++ b/src/core/MCPServer.ts
@@ -14,7 +14,7 @@ import {
 import { ToolProtocol } from '../tools/BaseTool.js';
 import { PromptProtocol } from '../prompts/BasePrompt.js';
 import { ResourceProtocol } from '../resources/BaseResource.js';
-import { readFileSync } from 'fs';
+import { readFileSync, existsSync } from 'fs';
 import { join, resolve, dirname } from 'path';
 import { logger } from './Logger.js';
 import { ToolLoader } from '../loaders/toolLoader.js';
@@ -103,9 +103,38 @@ export class MCPServer {
     if (configPath) {
       return configPath;
     }
-    if (process.argv[1]) {
-      return dirname(process.argv[1]);
+
+    // 1. Check project root dist/ directory (most common case)
+    const projectRoot = process.cwd();
+    const distPath = join(projectRoot, 'dist');
+    if (existsSync(distPath)) {
+      logger.debug(`Using project's dist directory: ${distPath}`);
+      return distPath;
     }
+
+    // 2. Walk up from the main module (process.argv[1]) to find dist/.
+    //    Handles npx where argv[1] is deep inside a temp/cache directory.
+    const mainModulePath = process.argv[1];
+    if (mainModulePath) {
+      let searchDir = dirname(mainModulePath);
+      for (let i = 0; i < 5; i++) {
+        const candidate = join(searchDir, 'dist');
+        if (existsSync(candidate)) {
+          logger.debug(`Found dist/ by walking up from argv[1]: ${candidate}`);
+          return candidate;
+        }
+        const parent = dirname(searchDir);
+        if (parent === searchDir) break;
+        searchDir = parent;
+      }
+
+      // 3. Fallback: use argv[1] dirname directly
+      const moduleDir = dirname(mainModulePath);
+      const basePath = moduleDir.endsWith('dist') ? moduleDir : join(moduleDir, 'dist');
+      logger.debug(`Using module path-based resolution: ${basePath}`);
+      return basePath;
+    }
+
     return process.cwd();
   }
 

